File: MCP服务器/windsurf/valstro_markdown-rules-mcp_fda82dd_typescript/src/logger.ts
import { config } from "./config.js";

const logLevel = config.LOG_LEVEL;

const logLevelMap = {
  silent: 0,
  debug: 1,
  info: 2,
  warn: 3,
  error: 4,
};

const currentLogLevel = logLevelMap[logLevel as keyof typeof logLevelMap] || logLevelMap.info;

export const logger = {
  log: (message: string) => {
    if (currentLogLevel <= logLevelMap.info) {
      console.error(`[MD-INFO] ${message}`);
    }
  },
  info: (message: string) => {
    if (currentLogLevel <= logLevelMap.info) {
      console.error(`[MD-INFO] ${message}`);
    }
  },
  debug: (message: string) => {
    if (currentLogLevel <= logLevelMap.debug) {
      console.error(`[MD-DEBUG] ${message}`);
    }
  },
  error: (message: string) => {
    if (currentLogLevel <= logLevelMap.error) {
      console.error(`[MD-ERROR] ${message}`);
    }
  },
  warn: (message: string) => {
    if (currentLogLevel <= logLevelMap.warn) {
      console.error(`[MD-WARN] ${message}`);
    }
  },
};

File: MCP服务器/llama/rainu_ask-mai_612d85e_go/internal/controller/logger.go
package controller

import (
	"github.com/wailsapp/wails/v2/pkg/logger"
	"log/slog"
)

type logWrapper struct {
	log *slog.Logger
}

func (l logWrapper) Print(message string) {
	l.log.With("level", "print").Info(message)
}

func (l logWrapper) Trace(message string) {
	l.log.With("level", "trace").Debug(message)
}

func (l logWrapper) Debug(message string) {
	l.log.With("level", "debug").Debug(message)
}

func (l logWrapper) Info(message string) {
	l.log.With("level", "info").Info(message)
}

func (l logWrapper) Warning(message string) {
	l.log.With("level", "warning").Warn(message)
}

func (l logWrapper) Error(message string) {
	l.log.With("level", "error").Error(message)
}

func (l logWrapper) Fatal(message string) {
	l.log.With("level", "fatal").Error(message)
}

func newDefaultLogger() logger.Logger {
	return &logWrapper{
		log: slog.With("source", "wails"),
	}
}
